jbpm-test-coverage/src/test/java/org/jbpm/test/functional/timer/BoundaryEventWithOutputSetPersistenceTest.java

```diff
@@ -0,0 +1,153 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.test.functional.timer;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.drools.core.process.instance.WorkItemHandler;
+import org.jbpm.test.JbpmTestCase;
+import org.junit.Before;
+import org.junit.Test;
+import org.kie.api.runtime.KieSession;
+import org.kie.api.runtime.manager.RuntimeEngine;
+import org.kie.api.runtime.manager.audit.VariableInstanceLog;
+import org.kie.api.runtime.process.ProcessInstance;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import junit.framework.Assert;
+
+/**
+ */
+public class BoundaryEventWithOutputSetPersistenceTest extends JbpmTestCase {
+
+    // General setup
+    private static final Logger logger = LoggerFactory.getLogger(BoundaryEventWithOutputSetPersistenceTest.class);
+
+    
+    private final static String PROCESS_FILE_NAME = "org/jbpm/test/functional/timer/BoundaryTimerProcessWithOutputSet.bpmn2";
+    private final static String PROCESS_NAME = "BoundaryTimerEventProcess";
+    private final static String WORK_ITEM_HANLDER_TASK = "Human Task";
+    
+    private final static String TIMER_FIRED_PROP = "timerFired";
+    private final static String TIMER_FIRED_TIME_PROP = "afterTimerTime";
+
+    public BoundaryEventWithOutputSetPersistenceTest() { 
+        super(true, true);
+    }
+    
+    @Before
+    public void setup() { 
+        System.clearProperty(TIMER_FIRED_PROP);
+        System.clearProperty(TIMER_FIRED_TIME_PROP);
+    }
+    
+    @Test
+    public void boundaryEventTimerAndCompleteHumanTask() throws InterruptedException {
+        /**
+         * First we set up everything and start the process
+         */
+        createRuntimeManager(PROCESS_FILE_NAME);
+        RuntimeEngine runtimeEngine = getRuntimeEngine();
+        KieSession ksession = runtimeEngine.getKieSession();
+        long ksessionId = ksession.getIdentifier();
+        assertTrue("session id not saved.", ksessionId > 0);
+        
+        HumanTaskMockHandler humanTaskMockHandler = new HumanTaskMockHandler();
+        ProcessInstance process = registerHTHandlerAndStartProcess(ksession, humanTaskMockHandler);
+        
+        sleepAndVerifyTimerRuns(process.getState());
+        List<? extends VariableInstanceLog> logs = getLogService().findVariableInstances(process.getId());
+        Assert.assertEquals(3, logs.size());
```

Comment:
as per interval in timer is set to `500ms###1s` and we wait for 2 seconds, expected is `4` instead of `3`.
Though I would try to increase interval period, to avoid having any potential flaky test based on timing, just in case, wdyt?

Reply:
it does not start in 0. but 500. anyway iwe are not testing that anymore. removed.